### PR TITLE
Move SwiftParser/Diagnostics to its own module

### PR DESCRIPTION
**Pull request by ahoppen**
Just a bit of cleanup I wanted to do to keep the modules small and focused.

The idea is that `SwiftParser` just creates a SwiftSyntax tree and `SwiftParserDiagnositcs` produces nice error messages for such a tree.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Package.swift [hunk 1]
```diff
@@ -124,14 +124,21 @@ let package = Package(
...
+    ),
     .target(
       name: "SwiftOperators",
       dependencies: ["SwiftSyntax", "SwiftParser", "SwiftDiagnostics"]
     ),
     .target(
       name: "SwiftCompilerSupport",
-      dependencies: [
-        "SwiftSyntax", "SwiftParser", "SwiftDiagnostics", "SwiftOperators"],
+      dependencies: ["SwiftSyntax", "SwiftParser", "SwiftParserDiagnostics",
```
Comment (DougGregor) [inline Package.swift:140]: You might need a paired PR to the Swift repository to link against `SwiftParserDiagnostics`.

**Review by DougGregor — APPROVED** [review]
I like this refactoring, because it keeps the parser module small and focused.

**Review by CodaFi — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by DougGregor — APPROVED** [review]
Yeah, I like this idea.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-stress-tester/pull/205

@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-stress-tester/pull/205

@swift-ci Please test

**Pull request merged**